pcsx2/GS/Renderers/SW/GSSetupPrimCodeGenerator.arm64.cpp

```diff
@@ -225,13 +225,16 @@ void GSSetupPrimCodeGenerator::Color()
 		// GSVector4 c = dscan.c;
 		armAsm->Ldr(v16, MemOperand(_dscan, offsetof(GSVertexSW, c)));
 
-		// m_local.d4.c = GSVector4i(c * 4.0f).xzyw().ps32();
+		// constexpr VectorI mask16 = VectorI::cxpr(0xFFFF);
+		armAsm->Movi(v17.V4S(), 0xFFFF);
 
+		// local.d4.c = (GSVector4i(dscan.c * step_shift) & mask16).xzyw().pu32();
 		armAsm->Fmul(v2.V4S(), v16.V4S(), v3.V4S());
 		armAsm->Fcvtzs(v2.V4S(), v2.V4S());
+		armAsm->And(v2.V4S(), v17.V4S());
 		armAsm->Rev64(_vscratch.V4S(), v2.V4S());
 		armAsm->Uzp1(v2.V4S(), v2.V4S(), _vscratch.V4S());
-		armAsm->Sqxtn(v2.V4H(), v2.V4S());
+		armAsm->Uqxtn(v2.V4H(), v2.V4S());
```

Comment:
```suggestion
		armAsm->Uzp1(v2.V8H(), v2.V8H(), v2.V8H());
```
Can drop the dup afterwards too